library/asn1parse.c

```diff
@@ -174,6 +180,9 @@ int mbedtls_asn1_get_mpi( unsigned char **p,
     if( ( ret = mbedtls_asn1_get_tag( p, end, &len, MBEDTLS_ASN1_INTEGER ) ) != 0 )
         return( ret );
 
+    if( ( ret = mbedtls_check_shortest_asn1_integer( *p, len ) ) != 0 )
+        return( ret );
+
```

Comment:
This part is related to the issue https://github.com/ARMmbed/mbedtls/issues/2099